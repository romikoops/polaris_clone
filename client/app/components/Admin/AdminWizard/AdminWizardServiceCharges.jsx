import React, { Component } from 'react'
import PropTypes from '../../../prop-types'
import styles from '../Admin.scss'
// import { AdminHubTile } from '../';
import { RoundButton } from '../../RoundButton/RoundButton'
// import {v4} from 'node-uuid';
import FileUploader from '../../../components/FileUploader/FileUploader'
import history from '../../../helpers'

export class AdminWizardServiceCharges extends Component {
  static back () {
    history.goBack()
  }
  constructor (props) {
    super(props)
    this.nextStep = this.nextStep.bind(this)
  }
  nextStep () {
    this.props.adminTools.goTo('/admin/wizard/pricings')
  }
  render () {
    const { theme, adminTools } = this.props
    const scUrl = '/admin/hubs/process_csv'
    const textStyle = {
      background:
        theme && theme.colors
          ? `-webkit-linear-gradient(left, ${theme.colors.primary},${theme.colors.secondary})`
          : 'black'
    }
    const backButton = (
      <div className="flex-none layout-row">
        <RoundButton
          theme={theme}
          size="small"
          text="Back"
          handleNext={AdminWizardServiceCharges.back}
          iconClass="fa-chevron-left"
        />
      </div>
    )
    const nextButton = (
      <div className="flex-none layout-row">
        <RoundButton
          theme={theme}
          size="small"
          active
          text="Next"
          handleNext={this.nextStep}
          iconClass="fa-chevron-right"
        />
      </div>
    )

    return (
      <div className="flex-100 layout-row layout-wrap layout-align-start-center">
        <div className="flex-100 layout-row layout-wrap layout-align-start-start">
          <div className={`flex-100 layout-row layout-align-start-center ${styles.sec_title}`}>
            <p className={` ${styles.sec_title_text} flex-none`} style={textStyle}>
              Service Charges
            </p>
          </div>
          <div
            className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_upload}`}
          >
            <p className="flex-none">Upload Service Charge Sheet</p>
            <FileUploader
              theme={theme}
              url={scUrl}
              dispatchFn={adminTools.wizardSCharge}
              type="xlsx"
              text="Hub .xlsx"
            />
          </div>
          <div className="layout-row flex-100 layout-wrap layout-align-start-center">
            {/* {hubList} */}
          </div>
          <div
            className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_title}`}
          >
            {backButton}
            {nextButton}
          </div>
        </div>
      </div>
    )
  }
}
AdminWizardServiceCharges.propTypes = {
  theme: PropTypes.theme,
  adminTools: PropTypes.shape({
    goTo: PropTypes.func
  }).isRequired
}

AdminWizardServiceCharges.defaultProps = {
  theme: null
}

export default AdminWizardServiceCharges
