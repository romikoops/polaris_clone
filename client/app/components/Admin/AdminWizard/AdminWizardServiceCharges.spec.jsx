import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity } from '../../../mock'
import AdminWizardServiceCharges from './AdminWizardServiceCharges'

const propsBase = {
  theme,
  adminTools: {
    wizardSCharge: identity
  }
}

test('shallow render', () => {
  expect(shallow(<AdminWizardServiceCharges {...propsBase} />)).toMatchSnapshot()
})
