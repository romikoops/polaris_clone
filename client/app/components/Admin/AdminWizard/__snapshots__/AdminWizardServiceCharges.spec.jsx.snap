// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render 1`] = `
<div
  className="flex-100 layout-row layout-wrap layout-align-start-center"
>
  <div
    className="flex-100 layout-row layout-wrap layout-align-start-start"
  >
    <div
      className="flex-100 layout-row layout-align-start-center sec_title"
    >
      <p
        className=" sec_title_text flex-none"
        style={
          Object {
            "background": "-webkit-linear-gradient(left, #333,#fafafa)",
          }
        }
      >
        Service Charges
      </p>
    </div>
    <div
      className="flex-100 layout-row layout-align-space-between-center sec_upload"
    >
      <p
        className="flex-none"
      >
        Upload Service Charge Sheet
      </p>
      <FileUploader
        dispatchFn={[Function]}
        formClasses=""
        size="small"
        square={false}
        t={[Function]}
        text="Hub .xlsx"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
        tooltip=""
        type="xlsx"
        uploadFn={null}
        url="/admin/hubs/process_csv"
      />
    </div>
    <div
      className="layout-row flex-100 layout-wrap layout-align-start-center"
    />
    <div
      className="flex-100 layout-row layout-align-space-between-center sec_title"
    >
      <div
        className="flex-none layout-row"
      >
        <RoundButton
          active={false}
          back={false}
          classNames=""
          disabled={false}
          flexContainer="100"
          handleDisabled={null}
          handleNext={[Function]}
          icon=""
          iconClass="fa-chevron-left"
          inverse={false}
          size="small"
          text="Back"
          theme={
            Object {
              "colors": Object {
                "primary": "#333",
                "secondary": "#fafafa",
              },
            }
          }
        />
      </div>
      <div
        className="flex-none layout-row"
      >
        <RoundButton
          active={true}
          back={false}
          classNames=""
          disabled={false}
          flexContainer="100"
          handleDisabled={null}
          handleNext={[Function]}
          icon=""
          iconClass="fa-chevron-right"
          inverse={false}
          size="small"
          text="Next"
          theme={
            Object {
              "colors": Object {
                "primary": "#333",
                "secondary": "#fafafa",
              },
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
